backend/cmd/kubeconfig.go

```diff
@@ -30,7 +32,28 @@ func GetContextsFromKubeConfigFile(kubeConfigPath string) ([]Context, error) {
 		}
 
 		authInfo := config.AuthInfos[value.AuthInfo]
-		cluster := Cluster{key, clusterConfig.Server, clusterConfig}
+		authProvider := authInfo.AuthProvider
+
+		var authConfig AuthConfig
+
+		if authProvider != nil {
+			authType := authProvider.Name
+			clientID := authProvider.Config["client-id"]
+			clientSecret := authProvider.Config["client-secret"]
+			idpIssuerURL := authProvider.Config["idp-issuer-url"]
+
+			authConfig.ClientID = clientID
+			authConfig.ClientSecret = clientSecret
+			authConfig.IdpIssuerURL = idpIssuerURL
+			authConfig.Type = authType
```

Comment:
You can assign each of these with the `authProvider.*` directly, i.e. no need for those temporary vars.